spec/features/admin_sessions_spec.rb

```diff
@@ -1,6 +1,10 @@
 require "rails_helper"
 
 RSpec.feature "Admin sessions" do
+  before do
+    allow_any_instance_of(DfeSignin::Authorisation).to receive(:has_role?) { true }
```

Comment:
By stubbing out the response of this class we lose integration coverage that would give us confidence the whole system works end-to-end. Ideally our feature specs would never stub our own code. If we need to stub anything, then it should be the underlying responses we get back from the auth calls. Have a look at `spec/support/omniauth.rb` to see where we're already doing that. It should be possible to modify the mock auth response there so that we get back an auth hash that looks like what we'd expect.

Reply:
I've fixed this in the sessions spec by using Webmock to stub the response from the DfE Signin API, but it seemed like overkill to repeat this work for every single login (I could have used hooks and done all the Webmock stub there, but I felt this was sacrificing readability/findability), so have kept the stubs in the other feature and request steps, as we're not _really_ testing the API response here, we just care if we're logged in or not